File: dotnet/DesignPatterns/StructuralDesignPatterns/BridgeDesignPattern/BridgeDesignPattern.Example1/Tv.cs
namespace BridgeDesignPattern.Example1;

public class Tv : IDevice
{
	public bool IsEnabled { get; private set; }
	public int Volume { get; set; } = 10;
	public int Channel { get; set; } = 1;
	public void Enable()
	{
		IsEnabled = true;
	}
	public void Disable()
	{
		IsEnabled = false;
	}
}

File: java/DesignPatterns/StructuralDesignPatterns/BridgeDesignPattern/BridgeDesignPattern.Example1/src/main/java/structural/bridge/example1/Tv.java
package structural.bridge.example1;

public class Tv implements IDevice {
    private boolean enabled;
    private int volume = 10;
    private int channel = 1;

    @Override public boolean isEnabled() { return enabled; }
    @Override public int getVolume() { return volume; }
    @Override public void setVolume(int volume) { this.volume = volume; }
    @Override public int getChannel() { return channel; }
    @Override public void setChannel(int channel) { this.channel = channel; }
    @Override public void enable() { enabled = true; }
    @Override public void disable() { enabled = false; }
}
